servers/rendering/rasterizer_rd/shaders/scene_high_end.glsl

```diff
@@ -1511,371 +1023,896 @@ void sdfgi_process(uint cascade, vec3 cascade_pos, vec3 cam_pos, vec3 cam_normal
+	return cluster_cell.y >> CLUSTER_COUNTER_SHIFT;
+}
+uint GET_SPOT_LIGHT(uvec4 cluster_cell, uint index) {
+	uint spot_light_pointer = cluster_cell.y & CLUSTER_POINTER_MASK;
+	uint light_index = cluster_data.indices[spot_light_pointer + index];
+	return light_index;
+}
+uint OMNI_LIGHT_COUNT(uvec4 cluster_cell) {
+	return cluster_cell.x >> CLUSTER_COUNTER_SHIFT;
+}
+uint GET_OMNI_LIGHT(uvec4 cluster_cell, uint index) {
+	uint omni_light_pointer = cluster_cell.x & CLUSTER_POINTER_MASK;
+	uint light_index = cluster_data.indices[omni_light_pointer + index];
+	return light_index;
+}
+uint DIRECTIONAL_LIGHT_COUNT() {
+	return scene_data.directional_light_count;
+}
+uint GET_DIRECTIONAL_LIGHT(uint index) {
+	return index;
+}
+uint REFLECTION_PROBE_COUNT(uvec4 cluster_cell) {
+	return cluster_cell.z >> CLUSTER_COUNTER_SHIFT;
+}
 
-	{ // process decals
+#else // !defined(MODE_RENDER_DEPTH) && !defined(MODE_UNSHADED)
 
-		uint decal_count = cluster_cell.w >> CLUSTER_COUNTER_SHIFT;
-		uint decal_pointer = cluster_cell.w & CLUSTER_POINTER_MASK;
+void REFLECTION_PROCESS(uvec4 cluster_cell, uint index, vec3 vertex, vec3 normal, float roughness, vec3 ambient_light, vec3 specular_light, inout vec4 ambient_accum, inout vec4 reflection_accum) {
+}
+bool SHOULD_RENDER_LIGHT(uint light_idx) {
+    return false;
+}
+bool SHOULD_RENDER_DIR_LIGHT(uint index) {
+    return false;
+}
+vec3 OMNI_PROJECTOR_PROCESS(uint light_idx, vec3 vertex, vec3 vertex_ddx, vec3 vertex_ddy) {
+    return vec3(1.0);
+}
+vec3 SPOT_PROJECTOR_PROCESS(uint light_idx, vec3 vertex, vec3 vertex_ddx, vec3 vertex_ddy) {
+    return vec3(1.0);
+}
+bool OMNI_SHADOW_PROCESS(uint light_idx, vec3 vertex, vec3 normal_interp, out float shadow, inout float transmittance_z) {
+    shadow = 1.0;
+    return false;
+}
+bool SPOT_SHADOW_PROCESS(uint light_idx, vec3 vertex, vec3 normal_interp, out float shadow, inout float transmittance_z) {
+    shadow = 1.0;
+    return false;
+}
+bool DIRECTIONAL_SHADOW_PROCESS(uint index, vec3 vertex, vec3 normal_interp, out vec3 shadow_color, out float shadow, inout float transmittance_z) {
+    shadow_color = vec3(0.0);
+    shadow = 1.0;
+    return false;
+}
 
-		//do outside for performance and avoiding arctifacts
+uint SPOT_LIGHT_COUNT(uvec4 cluster_cell) {
+	return 0;
+}
+uint GET_SPOT_LIGHT(uvec4 cluster_cell, uint index) {
+	return 0;
+}
+uint OMNI_LIGHT_COUNT(uvec4 cluster_cell) {
+	return 0;
+}
+uint GET_OMNI_LIGHT(uvec4 cluster_cell, uint index) {
+	return 0;
+}
+uint DIRECTIONAL_LIGHT_COUNT() {
+	return 0;
+}
+uint GET_DIRECTIONAL_LIGHT(uint index) {
+	return index;
+}
+uint REFLECTION_PROBE_COUNT(uvec4 cluster_cell) {
+	return 0;
+}
 
-		for (uint i = 0; i < decal_count; i++) {
-			uint decal_index = cluster_data.indices[decal_pointer + i];
-			if (!bool(decals.data[decal_index].mask & instances.data[instance_index].layer_mask)) {
-				continue; //not masked
-			}
+#endif //!defined(MODE_RENDER_DEPTH) && !defined(MODE_UNSHADED)
```

Comment:
Is this accurate?

Reply:
Not sure if this is referring to the green or the red, but the diff here is comparing two unrelated parts of code.

Reply:
Remove the comment if inaccurate?